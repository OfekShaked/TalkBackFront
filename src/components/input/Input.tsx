import React from 'react';
import { Button } from '@material-ui/core'

import './InputStyle.scss';

interface IInput {
    setMessage: any;
    sendMessage: any;
    message: any;
}
const Input = (props: IInput) => {
    const { setMessage, sendMessage, message } = props;
    return (
        <form className="form">
            <input
                className="input"
                type="text"
                placeholder="Type a message..."
                value={message}
                onChange={({ target: { value } }) => setMessage(value)}
                onKeyPress={event => event.key === 'Enter' ? sendMessage(event) : null}
            />
            <Button className="sendButton" onClick={e => sendMessage(e)}>Send</Button>
        </form>
    )
}

export default Input;