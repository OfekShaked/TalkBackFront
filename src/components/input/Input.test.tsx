import React from 'react';
import renderer from 'react-test-renderer';
import Input from './Input';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { mount, configure } from 'enzyme';
import { Button } from '@material-ui/core';

configure({ adapter: new Adapter() });

const sendMessage = jest.fn();
const setMessage = jest.fn();

describe('input tests', () => {
    it('renders input component', async () => {
        const component = renderer.create(
                <Input sendMessage={sendMessage} setMessage={setMessage} message={""}></Input>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('click on the send message button works', async () => {
        const component = mount(
            <Input sendMessage={sendMessage} setMessage={setMessage} message={""}></Input>
        );
        const btn = component.find(Button);
        btn.simulate("click");
        expect(sendMessage).toBeCalled();
    });

    it('click enter sends a message',async()=>{
        const component = mount(
            <Input sendMessage={sendMessage} setMessage={setMessage} message={""}></Input>
        );
        const input = component.find('input');
        input.simulate('keyPress', {
            key: 'Enter',
            keyCode: 13,
            which: 13,
        });
        expect(sendMessage).toBeCalled();
    })
    it('text change works',async()=>{
        const component = mount(
            <Input sendMessage={sendMessage} setMessage={setMessage} message={""}></Input>
        );
        const input = component.find('input');
        input.simulate('change', { target: { value: 'Hello' } })
        expect(setMessage).toBeCalledWith("Hello");
    })
    

})

