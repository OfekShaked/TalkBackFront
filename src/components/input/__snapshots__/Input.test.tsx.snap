// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input tests renders input component 1`] = `
<form
  className="form"
>
  <input
    className="input"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Type a message..."
    type="text"
    value=""
  />
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text sendButton"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Send
    </span>
  </button>
</form>
`;
